pipeline_kinesis.c

```diff
@@ -49,17 +49,22 @@
 PG_MODULE_MAGIC;
 
 extern void _PG_init(void);
-
 HTAB *consumer_info;
 
+#define MAX_PROCS 8
+
 /*
  * Shared state - used for starting / terminating workers
  */
 typedef struct KinesisConsumerInfo
 {
-	Oid oid;
+	int id;
 	Oid dboid;
-	BackgroundWorkerHandle handle;
+
+	int start_seq;
+	int num_procs;
+
+	BackgroundWorkerHandle handle[MAX_PROCS];
```

Comment:
You can use a seperate hash table for procs so we're not limited to 8. Like the way `KafkaConsumerProc` and `KafkaConsumerGroup` work.